### PR TITLE
fix: update pep508_rs which had unmaintained dependencies

### PR DESCRIPTION
**Pull request by volgar1x**

### CODE REVIEW

File: src/commands/add.rs [hunk 1]
```diff
@@ -68,13 +68,10 @@ fn merge_requirements(old: &Requirement, new: &Requirement) -> Option<Requiremen
     }
     merged.extras.extend(new.extras.iter().cloned());
     merged.extras.dedup();
-    if let Some(new_marker) = new.marker.as_ref() {
-        if old.marker.is_some() && old.marker.as_ref() != Some(new_marker) {
-            return None;
-        } else {
-            merged.marker = Some(new_marker.clone());
-        }
+    if old.marker != new.marker {
```
Comment (jpopesculian) [inline src/commands/add.rs:71]: this isn't exactly the same thing. the old code is saying if there exists a new marker, use that if there is no old marker or if its the same. the new code here wouldn't produce the same result. for example:

```rust
old.marker = None
new.marker = Some(marker)
// old code
merged.marker = Some(marker)
// new code
return None
```

File: src/commands/add.rs [hunk 2]
```diff
@@ -92,12 +89,7 @@ fn requirement_needs_update(old: &Requirement, new: &Requirement) -> bool {
             return true;
         }
     }
-    if let Some(new_marker) = new.marker.as_ref() {
-        if old.marker.as_ref() != Some(new_marker) {
-            return true;
-        }
-    }
-    false
+    old.marker == new.marker
```
Comment (jpopesculian) [inline src/commands/add.rs:92]: same principle here

**Review by jpopesculian — CHANGES_REQUESTED** [review]

File: src/commands/add.rs (hunk 1 above)
Comment (volgar1x) [reply src/commands/add.rs:71]: The `marker` field isn't an `Option<T>` anymore 🤐

**Review by jpopesculian — APPROVED** [review]
okie dokie! sorry I misunderstood. looks good. I'd like to replace this all with the uv builtins at some point. it doesn't make sense to reinvent the wheel and they do it better than us

**Pull request merged**